### PR TITLE
fix(service): apply bento project context to runners

### PR DESCRIPTION
**Pull request by bojiang**
## Description

revert #2713 

## Checklist:

- [ ] My code follows the bentoml code style, both `make format` and
  `make lint` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Review by aarnphm — COMMENTED** [review]
rmbr to remove the weakref object as well :)

File: bentoml/_internal/service/loader.py [hunk 1]
```diff
@@ -347,4 +347,7 @@ def load(
                     f"import bento in local store: `{e1}`, or if you are importing by "
                     f"python module path: `{e2}`"
                 )
+
+    if svc.bento:
+        on_load_bento(svc, svc.bento)
```
Comment (bojiang) [inline bentoml/_internal/service/loader.py:352]:
```suggestion
```

File: bentoml/_internal/bento/bento.py [hunk 2]
```diff
@@ -318,6 +319,14 @@ def validate(self):
...
         return f'Bento(tag="{self.tag}")'
 
+    @contextlib.contextmanager
+    def patch_context(self):
+        with BentoMLContainer.model_store.patch(self._model_store):
+            pwd = os.getcwd()
+            os.chdir(self.path)
+            yield
+            os.chdir(pwd)
+
```
Comment (ssheng) [inline bentoml/_internal/bento/bento.py:329]: Do we want to raise if another bento has already patched context for model store?

File: bentoml/_internal/service/service.py [hunk 3]
```diff
@@ -234,6 +232,13 @@ def on_load_bento(svc: Service, bento: Bento):
     object.__setattr__(svc, "bento", bento)
     object.__setattr__(svc, "tag", bento.info.tag)
 
+    import contextlib
+
+    exit_stack = contextlib.ExitStack()
+    exit_stack.enter_context(bento.patch_context())
+
+    object.__setattr__(svc, "_exit_stack", exit_stack)  # attach to the service object
+
```
Comment (parano) [inline bentoml/_internal/service/service.py:241]: add this to service class:

```python

def __del__(self):
    if self._exit_stack is not None:
        self._exit_stack.close()
```

**Review by parano — CHANGES_REQUESTED** [review]

**Comment by parano** [comment]
addressed in https://github.com/bentoml/BentoML/pull/2737

**Pull request closed without merging**